includes/abstracts/abstract-wc-product.php

```diff
@@ -1560,6 +1570,10 @@ public function backorders_require_notification() {
 	 * @return bool
 	 */
 	public function is_on_backorder( $qty_in_cart = 0 ) {
+		if ( 'onbackorder' === $this->get_stock_status() ) {
+			return true;
+		}
+
 		return $this->managing_stock() && $this->backorders_allowed() && ( $this->get_stock_quantity() - $qty_in_cart ) < 0 ? true : false;
```

Comment:
Is this logic needed anymore? Stock status can be taken care of on read so it's always accurate.

Reply:
I think it's still needed, but I don't feel very strongly about that. :) Let me know if you disagree. Here is why:

If the product allows backorders and the inventory goes below 0, `is_on_backorder` will return `true` even if the product has not been saved or freshly read from the DB.

Reply:
I don't follow what you mean... Won't the stock status be set by the data store before this gets called?

Reply:
1. Load product. Product is in stock and has inventory.
2. Decrease inventory below 0. Don't save yet.
3. Call `is_on_backorder`. It will return true, even though the product hasn't been saved and officially the stock status is 'instock'.

Seems like a useful feature because it can tell you whether you're backordered before you've saved. I might be overthinking it, though.

Reply:
I think so, what if the internal CRUD methods for decreasing stock should set it so the object data is all correct?

Reply:
Upon further research, we have to leave this method with that check. It handles the `$qty_in_cart` param which is needed during checkout to see if a product will be backordered.